packages/docs-theme/src/components/baseExample.tsx

```diff
@@ -16,11 +17,50 @@ export interface IBaseExampleProps {
  * Starter class for all React example components.
  * Examples and options are rendered into separate containers.
  */
-export class BaseExample<S> extends React.PureComponent<IBaseExampleProps, S> {
+export class BaseExample<S> extends React.Component<IBaseExampleProps, S> {
     /** Define this prop to add a className to the example container */
     protected className: string;
 
+    // Can't put this in state, because the state typing is generic.
+    private hasDelayedBeforeInitialRender = false;
+    private hasCompletedInitialRender = false;
+
+    public shouldComponentUpdate(nextProps: IBaseExampleProps, nextState: S & object) {
+        return (
+            // HACKHACK: Permit one redundant re-render after the inital delay. (This will be the first proper render of
+            // the component.)
+            (this.hasDelayedBeforeInitialRender && !this.hasCompletedInitialRender) ||
+            // Now, mimic PureComponent shouldComponentUpdate behavior:
+            !Utils.shallowCompareKeys(this.props, nextProps) ||
+            !Utils.shallowCompareKeys(this.state, nextState)
```

Comment:
can you use `super.shouldComponentUpdate` from `PureComponent`?

Reply:
@giladgray Don't think so, because I'm not allowed to specify a `shouldComponentUpdate` on a `PureComponent`.